libs/gl-client/src/signer/mod.rs

```diff
@@ -225,15 +230,55 @@ impl Signer {
                 if r.timestamp != 0 {
                     data.put_u64(r.timestamp);
                 }
-                data.put(&r.rune[..]);
+                let rune: Vec<u8> = r.rune.clone();
+                data.put(&rune[..]);
 
                 pk.verify(&data, &r.signature)
+                    .map_err(|e| anyhow!("signature verification failed: {}", e))?;
+
+                self.verify_rune(r.clone())
                     .map(|_| r)
-                    .map_err(|e| anyhow!("signature verification failed: {}", e))
+                    .map_err(|e| anyhow!("rune verification failed: {}", e))
             })
             .collect()
     }
 
+    /// Verifies that the public key of the request and the signers rune version
+    /// match the corresponding restrictions of the rune.
+    fn verify_rune(&self, request: crate::pb::PendingRequest) -> Result<(), anyhow::Error> {
```

Comment:
Notice that you probably want the decoded PB payload, otherwise you will have a hard time checking the rune against the undecoded commands :-)

Reply:
:+1: